src/Build/BackEnd/Components/Scheduler/Scheduler.cs

```diff
@@ -1870,6 +1870,11 @@ BuildEventContext NewBuildEventContext()
                 var parentConfiguration = configCache[parentRequest.BuildRequest.ConfigurationId];
                 return (buildRequestConfiguration, parentConfiguration);
             }
+
+            string ConcatenateGlobalProperties(BuildRequestConfiguration configuration)
+            {
+                return string.Join(";", configuration.GlobalProperties.GetCopyOnReadEnumerable().Select(p => $"{p.Name}={p.EvaluatedValue}"));
```

Comment:
This makes a copy of the properties, right? Why is that necessary; could you condense it to just selecting name and its value in the dictionary?

Also, nit: add a space after the semicolon. Makes it look cleaner.